cli/compose/template/template_test.go

```diff
@@ -81,3 +81,63 @@ func TestNonAlphanumericDefault(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, "ok /non:-alphanumeric", result)
 }
+
+func TestMandatoryVariableErrors(t *testing.T) {
+	testCases := []struct {
+		template      string
+		expectedError string
+	}{
+		{
+			template:      "not ok ${UNSET_VAR:?Mandatory Variable Unset}",
+			expectedError: "Mandatory Variable Unset",
+		},
+		{
+			template:      "not ok ${BAR:?Mandatory Variable Empty}",
+			expectedError: "Mandatory Variable Empty",
+		},
+		{
+			template:      "not ok ${UNSET_VAR:?}",
+			expectedError: "",
+		},
+		{
+			template:      "not ok ${UNSET_VAR?Mandatory Variable Unset",
+			expectedError: "Mandatory Variable Unset",
+		},
+		{
+			template:      "not ok ${UNSET_VAR?}",
+			expectedError: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		_, err := Substitute(tc.template, defaultMapping)
+		assert.Error(t, err)
+		assert.IsType(t, &InvalidTemplateError{tc.expectedError}, err)
```

Comment:
This should check the error message as well as the type.

Reply:
Thank you for the review. I have updated the PR. Also removed the `tc.expectedError` from the `InvalidTemplateError` as it seemed unnecessary.